presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -864,6 +872,25 @@ public void testSchemaEvolution()
         assertUpdate("DROP TABLE test_schema_evolution");
     }
 
+    @Test
+    public void testStructSchemaEvolution()
+            throws IOException
+    {
+        assertUpdate("CREATE TABLE struct_test(field0 ROW(a int, b int))");
```

Comment:
Thanks for adding a test. I have some suggestions to make the test a little easier to read.

- CREATE and INSERT queries can be combined using CREATE TABLE AS SELECT (CTAS)
- filePathOld -> oldFilePath
- filePathNew -> newFilePath
- getSinglePath -> getOnlyPath to match Guava naming
- MaterializedResult -> assertQuery

```
        getQueryRunner().execute("CREATE TABLE test_struct(x) AS SELECT CAST(ROW(1, 2) AS ROW(a int, b int)) AS x");
        getQueryRunner().execute("CREATE TABLE test_struct_add_column(x) AS SELECT CAST(ROW(1, 2, 3) AS ROW(a int, b int, c int)) AS x");
        Path oldFilePath = getOnlyPath("test_struct");
        Path newDirectoryPath = getOnlyPath("test_struct_add_column").getParent();
        Files.move(oldFilePath, Paths.get(newDirectoryPath.toString(), "old_file"), ATOMIC_MOVE);

        assertQuery("SELECT * FROM test_struct_add_column", "SELECT (1, 2, 3) UNION ALL SELECT (1, 2, null)");
```